src/handlers/testing/exec/index.ts

```diff
@@ -102,39 +85,104 @@ class RunManager {
   }
 
   private async post<T>(path: string, body?: unknown): Promise<T> {
-    const resp = await fetch(`https://api.autoblocks.ai${path}`, {
+    const subpath = this.isCI ? '/testing/ci' : '/testing/local';
+    const resp = await fetch(`https://api.autoblocks.ai${subpath}${path}`, {
       method: 'POST',
       body: body ? JSON.stringify(body) : undefined,
       headers: {
         'Content-Type': 'application/json',
         Authorization: `Bearer ${this.apiKey}`,
       },
     });
+    const data = await resp.json();
     if (!resp.ok) {
-      throw new Error(
-        `POST ${path} failed: ${resp.status} ${await resp.text()}`,
-      );
+      throw new Error(`POST ${subpath}${path} failed: ${JSON.stringify(data)}`);
     }
-    return resp.json();
+    return data;
+  }
+
+  private get isCI(): boolean {
+    return process.env.CI === 'true';
+  }
+
+  async makeCIContext(): Promise<CIContext | undefined> {
+    if (!this.isCI) {
+      return undefined;
+    }
+    if (!this.ciContext) {
+      try {
+        this.ciContext = await makeCIContext();
```

Comment:
I initially thought this would cause function naming collisions - but I see this works because the `maxCIContext` you are defining here is a instance member and what you are calling on line 114 is an import.

Reply:
Ah yeah I see, any rename suggestions are welcome 😁